### PR TITLE
Sponsor block

### PR DESCRIPTION
**Pull request by vedanshujain**

### CODE REVIEW

File: public_html/wp-content/mu-plugins/blocks/assets/src/shared/block-content/index.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+/**
+ * Renderes a features image for a post.
+ *
+ * @param post
+ * @param imageClass
+ * @param defaultImageClass
+ * @returns {*}
+ * @constructor
+ */
+export function WCPTFeaturedImage( { post, imageClass, defaultImageClass } ) {
```
Comment (coreymckrill) [inline public_html/wp-content/mu-plugins/blocks/assets/src/shared/block-content/index.js]: Good idea turning this into a shared component. Given that it's part of a module that can be imported/exported, I don't think it needs any kind of namespacing prefix. I'd probably go with just `FeaturedImage`.

File: public_html/wp-content/mu-plugins/blocks/assets/src/sponsors/block-controls.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies.
+ */
+import Placeholder from 'react-select/src/components/Placeholder';
```
Comment (coreymckrill) [inline public_html/wp-content/mu-plugins/blocks/assets/src/sponsors/block-controls.js]: There is a Placeholder component in Core's `wp.components`. Did you mean to use the one from `react-select`?

File: public_html/wp-content/mu-plugins/blocks/assets/src/sponsors/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * Internal dependencies.
+ */
+import { edit } from './edit.js';
+
+export const name = 'wordcamp/sponsors';
+
+export const settings = {
+	title       : __( 'Sponsors', 'wordcamporg' ),
+	description : __( "We wouldn't have WordCamp without their support.", 'wordcamporg' ),
+	icon        : __( 'list-view', 'wordcamporg' ),
```
Comment (coreymckrill) [inline public_html/wp-content/mu-plugins/blocks/assets/src/sponsors/index.js]: For the Sponsors block I think this is supposed to be `heart`.

File: public_html/wp-content/mu-plugins/blocks/includes/sessions.php [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+use WordCamp\Blocks;
+
+/**
+ * Register block types and enqueue scripts.
+ *
+ * @return void
+ */
+function init() {
+	register_block_type(
+		'wordcamp/sessions',
```
Comment (coreymckrill) [inline public_html/wp-content/mu-plugins/blocks/includes/sessions.php]:
```suggestion
		'wordcamp/sponsors',
```

**Comment by vedanshujain** [comment]
Closing in favour of https://github.com/WordPress/wordcamp.org/pull/33

**Pull request closed without merging**